email_handler.py

```diff
@@ -536,6 +536,55 @@ def handle_email_sent_to_ourself(alias, from_addr: str, msg: Message, user):
     )
 
 
+def apply_dmarc_policy(alias: Alias, contact: Contact, msg: Message) -> Optional[str]:
+    spam_result = msg.get_all(headers.SPAMD_RESULT)
```

Comment:
I think it'd be more clear to extract a method `get_dmarc_status` that can return either one of the symbols in https://rspamd.com/doc/modules/dmarc.html#dmarc-in-rspamd. The rspamd symbols can also be stored as an enum. We can then create unit tests for this method with different rspamd cases.

Reply:
I wonder if we can get the dmarc info for all messages and send the following dmarc metrics to newrelic (via `newrelic.agent.record_custom_metric`)? This will help us to know how many messages have issues with dmarc.

```
Custom/DMARC_BAD_POLICY
Custom/DMARC_NA
Custom/DMARC_POLICY_ALLOW
Custom/DMARC_POLICY_REJECT
Custom/DMARC_POLICY_QUARANTINE
Custom/DMARC_POLICY_SOFTFAIL
```

Reply:
Will do.

Reply:
Done

Reply:
Added as a custom_event so we can see each dmarc result independently.